### PR TITLE
Hop upgrade to 0.7.0

### PR DESCRIPTION
**Pull request by joesmolsky**
Upgrade hop-client to 0.7.0.  This also involves specifying ads-streaming==2.3.0 and confluent-kaka==1.9.2 in the requirements.  This closes <https://github.com/SNEWS2/SNEWS_Publishing_Tools/issues/61>.

Also include is a change to unit test workflows.  Python versions 3.7-3.10 now run in series on push or pull to the main branch while version 3.9 is tested on pushes to any branch.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,23 +1,25 @@
-click>=8.0.4
+click~=8.1.2
```
Comment (KaraMelih) [inline requirements.txt:1]: We changed this and some others to more flexible `>=` following Spencer's PR. I think it was needed for Python 3.6 version. Although IceCube also upgraded, I don't know if we will ever need py3.6

**Review by KaraMelih — COMMENTED** [review]
We can think of which of the `~=` or `>=` is better for the requirements. For older python versions, newer versions of some packages do not work.

File: requirements.txt (hunk 1 above)
Comment (joesmolsky) [reply requirements.txt:1]: I think we are not going to support python 3.6 since IceCube no longer requires it there have been no other requests for it.

Comment (KaraMelih) [reply requirements.txt:1]: Okay. We also had this one for py3.6 compatibility;
https://github.com/SNEWS2/SNEWS_Publishing_Tools/blob/05e02890a9223debea4b7745f47a06dae005b5af/snews_pt/snews_format_checker.py#L3-L6

I guess we also don't need that anymore but I also think that it doesn't hurt to keep it.

**Review by KaraMelih — APPROVED** [review]

**Pull request merged**